include/macros.h

```diff
@@ -199,4 +199,10 @@ extern GraphicsContext* __gfxCtx;
                        ((height)-1) << G_TEXTURE_IMAGE_FRAC);                                                          \
     } while (0)
 
+#define OVERLAY_RELOCATION_OFFSET(overlayEntry) \
+    ((uintptr_t)((overlayEntry)->vramStart) - (uintptr_t)((overlayEntry)->loadedRamAddr))
+#define VRAM_PTR_SIZE(entry) ((uintptr_t)((entry)->vramEnd) - (uintptr_t)((entry)->vramStart))
+
+#define RELOCATE_ADDR(addr, vRamStart, allocu32) ((addr) - (vRamStart) + (allocu32))
```

Comment:
are these macros needed? I feel like its just hiding implementation details in a function where you would want to see these details up front

Reply:
Here is my opinion after reading the discussion in discord:
In general I agree with the others mentioning how these macros hide details in an already internal function. More specifically, I think `OVERLAY_RELOCATION_OFFSET` could just be replaced with a comment like Dragorn suggested, `VRAM_PTR_SIZE` feels unnecessary since doing `end - start` with these casts is very straightforward, and `RELOCATE_ADDR` isn't used so we probably don't need it (at least in oot)

Reply:
@hensldm wanted to try out using the same macro for both cases, instead of having two seperate ones. if that works out id probably be ok with a compromise to just have the one macro.
But overall I do still feel that its hiding stuff that we dont need to hide

Reply:
Oh right, thanks for the reminder.
So it doesn't look like I can combine them into 1 macro, loadfragment in MM doesn't like the `intptr_t` cast, but it seems necessary in `DLF` and the other common usages. 
However I can make `RELOCATE_ADDR` usable in both places (again it doesn't apply here in relocation.c yet as the function got refactored around MM pal time, and so the macro doesn't really work, though I do think it is very much an improvement in the old version). I've also added an `UNRELOCATE_ADDR` for the second function, though not particularly fond of the name.
https://github.com/zeldaret/oot/compare/master...hensldm:oot:dlf